pkg/virt-handler/vm.go

```diff
@@ -2560,6 +2565,13 @@ func (d *VirtualMachineController) hotplugSriovInterfaces(vmi *v1.VirtualMachine
 		return nil
 	}
 
+	rateLimitedExecutor := d.sriovHotplugExecutorPool.LoadOrStore(vmi.UID)
```

Comment:
Why don't you use - `vendor/k8s.io/client-go/util/workqueue/rate_limiting_queue.go`?
same as the [controller](https://github.com/kubevirt/kubevirt/blob/377e83878336f5d20f67336e530e2831486239ea/pkg/virt-handler/vm.go#L172). 
`workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "sriov")`

The `workqueue.DefaultControllerRateLimiter` is `ItemExponentialFailureRateLimiter`.

Reply:
I reviewed its implementation and tried to change the code to use it, I came up to a point where its basically a controller the requires another gorutine.
With more detailed, using the workqueue in this case includes  dequeue an element, do the hotplug and according to its result add the element back to the queue with rate-limiter.
Dequeue'ing an element is done with Get()  which is a blocking func (until it ables to dequeue an element).